### DISCUSSION TITLE
How to make ajax update the history cache

### DISCUSSION THREAD

**Discussion by baseplate-admin** [bigskysoftware/htmx#2116] [Q&A]
Hi, i have a js function that looks like this :

```ts
export async function goto({ url, verb, target }: { url: string; verb?: "GET" | "POST"; target: string }): Promise<void> {
    if (!verb) {
        verb = "GET";
    }
    await ajax(verb, url, target);

}
```

How to make it update the history cache?

**Comment by nickchomey** [comment]
It seems like you're not using htmx, which should do the Ajax and history stuff automatically...

**Reply by baseplate-admin** [reply]
> It seems like you're not using htmx, which should do the Ajax and history stuff automatically... 

Yes, `htmx` does everything automatically. But i am looking to use that functionality programatically. ( like from inside `svelte` )
I am using the ajax, more specifically this function

https://github.com/bigskysoftware/htmx/blob/89b523cde0c8f7b40baaaa4e76a2f25a36b42439/src/htmx.d.ts#L14-L52

My requirements are more specific in nature, I am using svelte to compile everything to web-component, then using the `htmx` router from there.

**Reply by nickchomey** [reply]
Apologies. Someone else will have to comment on how to do this - or look at the single htmx source code file and find where it handles history updates. You could even add breakpoints and test it in the browser to find exactly what is happening

**Reply by baseplate-admin** [reply]
> Apologies. Someone else will have to comment on how to do this - or look at the single htmx source code file and find where it handles history updates. You could even add breakpoints and test it in the browser to find exactly what is happening

Correct. I did that, and it seems that `htmx` doesn't have a public api to do this
